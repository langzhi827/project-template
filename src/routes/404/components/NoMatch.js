/*
 * @Author: harry.lang 
 * @Date: 2018-03-20 14:54:22 
 * @Last Modified by: harry.lang
 * @Last Modified time: 2019-01-02 15:32:17
 */
import React, { PureComponent } from 'react';
import { DatePicker, Dropdown, Menu, Icon } from 'antd';
import CSSModules from 'react-css-modules';

import styles from '../assets/noMatch.css';
import img from '../assets/images/hehe.png';

class NoMatch extends PureComponent {

    componentDidMount() {
        const { getList } = this.props;
        getList();
    }

    changeTitle() {
        this.props.updateTitle('404-' + (+new Date));
    }

    render() {
        const { nomatch } = this.props;

        const menu = (
            <Menu>
                <Menu.Item>
                    <a target="_blank" rel="noopener noreferrer" href="http://www.alipay.com/">1st menu item</a>
                </Menu.Item>
                <Menu.Item>
                    <a target="_blank" rel="noopener noreferrer" href="http://www.taobao.com/">2nd menu item</a>
                </Menu.Item>
                <Menu.Item>
                    <a target="_blank" rel="noopener noreferrer" href="http://www.tmall.com/">3rd menu item</a>
                </Menu.Item>
            </Menu>
        );

        return <div styleName="no-match">
            <h2>{nomatch.title}</h2>

            <DatePicker />
            <Dropdown overlay={menu}>
                <a className="ant-dropdown-link" href="#">
                    Hover me <Icon type="down" />
                </a>
            </Dropdown>

            <img styleName="img" src={img} alt="404" onClick={this.changeTitle.bind(this)} />

            <h3>No match for <code>{this.props.location.pathname}</code></h3>
        </div>;
    }
}

export default CSSModules(NoMatch, styles);
