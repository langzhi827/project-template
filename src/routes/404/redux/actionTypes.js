export default {
    UPDATE_TITLE: 'NOMATCH/UPDATE_TITLE',
    UPDATE_LIST: 'NOMATCH/UPDATE_LIST'
};
