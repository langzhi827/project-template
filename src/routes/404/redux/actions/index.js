/**
 *  Author: harry.lang
 *  Date: 2017/11/24
 *  Description: Created by harrylang on 2017/11/24.
 */
import constants from '../constants';
import fetch from '@/utils/fetch';
import api from '@/config/api';

/**
 * 修改title
 * @param title 新的标题
 * @returns {{type: string, payload: *}}
 */
export function updateTitle(title) {
    return {
        type: constants.UPDATE_TITLE,
        payload: title
    };
}

/**
 * 更新列表
 * @param list
 * @returns {{type: string, payload: *}}
 */
export const updateList = (list) => {
    return {
        type: constants.UPDATE_LIST,
        payload: list
    };
};

/**
 * 获取列表
 * @param params
 * @returns {Function}
 */
export const getList = (params) => {
    return async dispatch => {
        const result = await fetch(api.list);
        dispatch(updateList(result));
    };
};
