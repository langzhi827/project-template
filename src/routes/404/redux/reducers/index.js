/**
 *  Author: harry.lang
 *  Date: 2017/11/24
 *  Description: Created by harrylang on 2017/11/24.
 */
import constants from '../constants';

const action_handles = {
    [constants.UPDATE_TITLE]: (state, action) => {
        const title = action.payload;
        return Object.assign({}, state, { title: title });
    },
    [constants.UPDATE_LIST]: (state, action) => {
        const list = action.payload;
        return Object.assign({}, state, { list: list });
    }
};
const initialState = {
    title: '404',
    list: []
};

export default function (state = initialState, action) {
    const handler = action_handles[action.type];
    return handler ? handler(state, action) : state;
}

