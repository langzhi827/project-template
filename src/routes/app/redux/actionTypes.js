/**
 *  Author: harry.lang
 *  Date: 2017/11/24
 *  Description: Created by harrylang on 2017/11/24.
 */

export default {
    UPDATE_TITLE: 'APP/UPDATE_TITLE',
    UPDATE_LIST: 'APP/UPDATE_LIST'
};