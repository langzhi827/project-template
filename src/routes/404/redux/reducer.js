/**
 *  Author: harry.lang
 *  Date: 2017/11/24
 *  Description: Created by harrylang on 2017/11/24.
 */
import actionTypes from './actionTypes';

const action_handles = {
    [actionTypes.UPDATE_TITLE]: (state, action) => {
        const title = action.payload;
        return Object.assign({}, state, { title: title });
    },
    [actionTypes.UPDATE_LIST]: (state, action) => {
        const list = action.payload;
        return Object.assign({}, state, { list: list });
    }
};
const initialState = {
    title: '404',
    list: []
};

export default function (state = initialState, action) {
    const handler = action_handles[action.type];
    return handler ? handler(state, action) : state;
}

