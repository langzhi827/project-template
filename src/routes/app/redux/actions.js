/**
 *  Author: harry.lang
 *  Date: 2017/11/24
 *  Description: Created by harrylang on 2017/11/24.
 */
import actionTypes from './actionTypes';
import fetch from '@/utils/fetch';
import api from '@/config/api';

/**
 * 修改title
 * @param title 新的标题
 * @returns {{type: string, payload: *}}
 */
export function updateTitle(title) {
    return {
        type: actionTypes.UPDATE_TITLE,
        payload: title
    };
}

/**
 * 获取列表
 * @param params
 * @returns {Function}
 */
export function getList(params) {
    return async dispatch => {
        //return fetch(ENV.API.LIST).then(function (result) {
        //    dispatch(updateList(result))
        //});
        const result = await fetch(api.list + '?test=1', {
            params: {
                test1: 2
            }
        });
        dispatch(updateList(result));
    };
}

/**
 * 更新列表
 * @param list
 * @returns {{type: string, payload: *}}
 */
export function updateList(list) {
    return {
        type: actionTypes.UPDATE_LIST,
        payload: list
    };
}