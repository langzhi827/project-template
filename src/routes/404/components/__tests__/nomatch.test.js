import React from 'react';
import { render, mount, shallow } from 'enzyme';
import NoMatch from '../NoMatch';

import { createStore, applyMiddleware, combineReducers } from 'redux';
import { Provider } from 'react-redux';
import reducer from '../../redux/reducer';
import thunk from 'redux-thunk';

jest.useFakeTimers();

describe('NoMatch', () => {

    const initialState = {
        noMatch: {
            title: '404',
            list: []
        }
    }

    const middleware = [thunk];
    const store = createStore(
        combineReducers({
            noMatch: reducer
        }),
        initialState,
        applyMiddleware(...middleware)
    );

    const props = {
        location: { pathname: '/404' }
    };

    describe('renders correctly', () => {
        const wrapper = render(
            <Provider store={store}>
                <NoMatch {...props} />
            </Provider>
        );

        it('snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('default title', () => {
            expect(wrapper.find('h2').text()).toEqual(initialState.noMatch.title);
        });
    })

    describe('mount correctly', () => {
        const wrapper = mount(
            <Provider store={store} >
                <NoMatch  {...props} />
            </Provider>
        );

        it('snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('throw exception', () => {
            expect(() => wrapper).not.toThrow();
        });

        it('click img', () => {
            wrapper.find('img').simulate('click');
            expect(wrapper.find('h2').text()).toEqual(store.getState().noMatch.title);
        });
    })
});