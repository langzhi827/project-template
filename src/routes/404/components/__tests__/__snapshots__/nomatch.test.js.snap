// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch mount correctly snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(NoMatch)
    location={
      Object {
        "pathname": "/404",
      }
    }
  >
    <NoMatch
      dispatch={[Function]}
      getList={[Function]}
      location={
        Object {
          "pathname": "/404",
        }
      }
      title="404"
      updateTitle={[Function]}
    >
      <div
        className="noMatch"
      >
        <h2>
          404
        </h2>
        <img
          alt="404"
          className="img-2q9nM"
          onClick={[Function]}
          src="hehe.png"
        />
        <h3>
          No match for 
          <code>
            /404
          </code>
        </h3>
      </div>
    </NoMatch>
  </Connect(NoMatch)>
</Provider>
`;

exports[`NoMatch renders correctly snapshot 1`] = `
<div
  class="noMatch"
>
  <h2>
    404
  </h2>
  <img
    alt="404"
    class="img-2q9nM"
    src="hehe.png"
  />
  <h3>
    No match for 
    <code>
      /404
    </code>
  </h3>
</div>
`;
